server/src/main/java/org/opensearch/cluster/routing/allocation/RebalanceConstraints.java

```diff
@@ -31,7 +31,7 @@ public class RebalanceConstraints {
 
     public RebalanceConstraints(RebalanceParameter rebalanceParameter) {
         this.constraints = new HashMap<>();
-        this.constraints.put(INDEX_PRIMARY_SHARD_BALANCE_CONSTRAINT_ID, new Constraint(isPerIndexPrimaryShardsPerNodeBreached()));
+        this.constraints.put(INDEX_PRIMARY_SHARD_BALANCE_CONSTRAINT_ID, new Constraint(isPerIndexPrimaryShardsPerNodeBreached(0.0f)));
```

Comment:
Don't we need to pass the value of setting here?

Reply:
We are only allowing the buffer in allocation , but not in rebalance , as we want that to happen perfectly on an index level .